tools/accounts_cli.go

```diff
@@ -0,0 +1,259 @@
+	app.Usage = "Internal tool to add update and find users by role type"
+	app.Version = "0.0.1"
+	app.Author = "Jamie"
+	app.Email = "jamie@tidepool.org"
+
+	app.Commands = []cli.Command{
+
+		//e.g. update -e testie@user.org -r clinic
+		{
+			Name:      "update",
+			ShortName: "u",
+			Usage:     `update and existing account to have a role`,
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:  "userid",
+					Usage: "userid of the account your updating",
+				},
+				cli.StringFlag{
+					Name:  "role",
+					Value: "clinic",
+					Usage: "the role that is being given to the user, defaults to `clinic`",
+				},
+				cli.StringFlag{
+					Name:  "env",
+					Usage: "the environment we are running against",
+				},
+			},
+			Action: updateAccount,
+		},
+		//e.g. find -r clinic
+		{
+			Name:      "find-account",
+			ShortName: "fa",
+			Usage:     `find an account to be updated`,
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:  "email",
+					Usage: "eamil address of the account your finding",
+				},
+				cli.StringFlag{
+					Name:  "env",
+					Usage: "the environment we are running against",
+				},
+			},
+			Action: findAccount,
+		},
+		//e.g. find -r clinic
+		{
+			Name:      "find",
+			ShortName: "f",
+			Usage:     `find all accounts associated with a specific role`,
+			Flags: []cli.Flag{
+				cli.StringFlag{
+					Name:  "role",
+					Value: "clinic",
+					Usage: "the role that is being given to the user",
+				},
+				cli.StringFlag{
+					Name:  "env",
+					Usage: "the environment we are running against",
+				},
+			},
+			Action: findAccounts,
+		},
+	}
+
+	app.Run(os.Args)
+
+}
+
+func setHost(targetEnv string) string {
+
+	targetEnv = strings.ToLower(targetEnv)
+
+	fmt.Println("targeting environment ...", targetEnv)
+
+	if targetEnv == "dev" {
+		return dev_host
+	} else if targetEnv == "prd" {
+		return prd_host
+	} else if targetEnv == "stg" {
+		return stg_host
+	} else if targetEnv == "stg" {
+		return stg_host
+	}
+	return local_host
+}
+
+func (a *admin) login() {
+
+	urlPath := a.env + "/auth/serverlogin"
+
+	un := "hydrophone-local"
+	pw := os.Getenv("SERVER_SECRET")
+
+	if un == "" || pw == "" {
+		log.Fatal("username or password not set")
+	}
+
+	req, _ := http.NewRequest("POST", urlPath, nil)
+	req.Header.Add(tp_server_name, un)
+	req.Header.Add(tp_server_secret, pw)
+
+	res, err := a.client.Do(req)
+	if err != nil {
+		log.Fatal(fmt.Sprint("Login request failed", err.Error()))
+	}
+
+	switch res.StatusCode {
+	case 200:
+		a.token = res.Header.Get(tp_token)
+		return
+	default:
+		log.Fatal(fmt.Sprint("Login failed", res.StatusCode))
+		return
+	}
+}
+
+func findAccounts(c *cli.Context) {
+
+	hostEnv := setHost(c.String("env"))
+
+	adminUser := &admin{env: hostEnv, client: &http.Client{}}
+	adminUser.login()
+
+	roles := []byte(fmt.Sprintf(`{"roles":["%s"]}`, c.String("role")))
+
+	req, _ := http.NewRequest("GET", fmt.Sprintf("%s/auth/users", hostEnv), bytes.NewBuffer(roles))
```

Comment:
Usually GET requests don't have bodies. Instead you'd add you parameters to the query string.